docs/DeveloperGuide.adoc

```diff
@@ -153,98 +173,236 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-image::UndoRedoState2.png[]
+3. Refresh --
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+   Command: refresh
+
+Refreshes the list of due soon tasks. This command is useful for a situation where due soon list is not up to date due to a drastic change in time.
+
+==== Example Usage Scenario
+Given below is an example usage scenario and how the due soon list mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time.
+The list of tasks due soon will be initialized and displayed with the initial address book state. Initialization is done by filtering tasks that are due soon from the general list of tasks.
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Activity diagram:
 
-image::UndoRedoState3.png[]
+image::AddDueSoonActivityDiagram.png[width = "200", length = "300"]
+
+Step 2. The user executes `add` command to add a task.
+The `add` command execution checks if this task belongs in the due soon tasks list (i.e it is due in 7 days). If it does, the task is added to both lists, else it is only added to the main list.
+
+Step 3. The user executes `delete` command to delete a task.
+The `delete` command execution also checks if this task exists in the due soon task list. If it does, task is deleted from both lists, else it is only deleted from the main list.
+
+Step 4. The user executes `due soon` command to refresh the due soon task list.
+The `due soon` command execution will traverse the main list and check if there are any inconsistencies. For example, if there has been a change in time and a task is missing from the due soon task list, it will be added.
 
 [NOTE]
-If the `currentStatePointer` is at index 0, pointing to the initial address book state, then there are no previous address book states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the undo.
+If there are no such inconsistencies, the due soon command will not make any changes.
+
+Activity diagram:
 
-The following sequence diagram shows how the undo operation works:
+image::DueSoonActivityDiagram.png[width = "400", length = "600"]
 
-image::UndoSequenceDiagram.png[]
+Sequence diagram:
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+image::DueSoonSequenceDiagram.png[width = "600", length = "500"]
```

Comment:
1. Is the group still using AddressBook model? If not, remember to remove it / rename it appropriately from your sequence diagram!

2. Also, is there any arguments taken in by the functions? E.g. execute(), addDueSoonTask

3. For the cross that indicates a deletion, remember to place it a further ahead (not directly on the end of the lifeline bar)

4. Consider color coding the different sections (e.g. model, logic, ui) for better readability